### PR TITLE
engine: update to cli@3.17.0-beta and cleanup legacy APIs

### PR DESCRIPTION
**Pull request by bassosimone**
We must use check-in now instead of using the URL list API, which has been removed from the codebase.

While there, remove code that was unused.

Part of https://github.com/ooni/probe/issues/2273.

See https://github.com/ooni/probe/issues/2390.

### CODE REVIEW

File: app/src/main/java/org/openobservatory/ooniprobe/test/TestAsyncTask.java [hunk 1]
```diff
@@ -167,25 +160,35 @@ private void downloadURLs() {
...
             OONIContext ooniContext = session.newContextWithTimeout(30);
 
-            ThirdPartyServices.addLogExtra("ooniContext", app.getGson().toJson(ooniContext));
-
-            session.maybeUpdateResources(ooniContext);
-            OONIURLListConfig config = new OONIURLListConfig();
-            config.setCategories(app.getPreferenceManager().getEnabledCategoryArr().toArray(new String[0]));
+            /* TODO(bassosimone,aanorbel): figure out how to initialize onWifi
+                and charging by inspecting ServiceUtil.java
+             */
```
Comment (bassosimone) [inline app/src/main/java/org/openobservatory/ooniprobe/test/TestAsyncTask.java:165]: @aanorbel what do you think is the best way to do this? (Ideally we'd like to factor some common code?)

File: engine/src/main/java/org/openobservatory/engine/OONICheckInConfig.java [hunk 2]
```diff
@@ -55,7 +55,7 @@ public OONICheckInConfig(String softwareName,
         this.softwareVersion = softwareVersion;
         webConnectivity = new CheckInConfigWebConnectivity();
         for (String category : categories) {
-            this.webConnectivity.add(category);
+            this.webConnectivity.addCategory(category);
```
Comment (aanorbel) [inline engine/src/main/java/org/openobservatory/engine/OONICheckInConfig.java:58]: Change from `add` to `addCategory` https://github.com/ooni/probe-cli/blob/master/pkg/oonimkall/session.go#L346 not available in engine . Causing build failure

**Review by aanorbel — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/openobservatory/ooniprobe/common/ResubmitTask.java [hunk 3]
```diff
@@ -133,11 +133,6 @@ protected Boolean doInBackground(Integer... params) {
                     EngineProvider.get().getDefaultSessionConfig(
                             getActivity(), BuildConfig.SOFTWARE_NAME, BuildConfig.VERSION_NAME, logger, proxy)
             );
-            // Updating resources with no timeout because we don't know for sure how much
-            // it will take to download them and choosing a timeout may prevent the operation
-            // to ever complete. (Ideally the user should be able to interrupt the process
-            // and there should be no timeout here.)
-            session.maybeUpdateResources(session.newContext());
```
Comment (bassosimone) [inline app/src/main/java/org/openobservatory/ooniprobe/common/ResubmitTask.java:140]: The `maybeUpdateResources` API call has been a stub for quite some time. Removing it is harmless.

File: app/src/main/java/org/openobservatory/ooniprobe/domain/GenerateAutoRunServiceSuite.java [hunk 4]
```diff
@@ -46,8 +46,6 @@ public AbstractSuite generate(
                     )
             );
             OONIContext ooniContext = session.newContextWithTimeout(30);
-            session.maybeUpdateResources(ooniContext);
-
```
Comment (bassosimone) [inline app/src/main/java/org/openobservatory/ooniprobe/domain/GenerateAutoRunServiceSuite.java:50]: The `maybeUpdateResources` API call has been a stub for quite some time. Removing it is harmless.

**Review by aanorbel — APPROVED** [review]
LGTM

**Pull request merged**